aten/src/ATen/native/cudnn/Conv.cpp

```diff
@@ -813,8 +824,12 @@ void raw_cudnn_convolution_forward_out(
   // wasteful; we'd rather reuse the workspace.  OTOH, legacy group
   // convolution support is already pretty slow, so this might not
   // matter.  (This applies to raw_cudnn_convolution_backward_input as well.)
-  cudnnConvolutionFwdAlgo_t fwdAlg;
-  Workspace workspace = chooseAlgorithm(args, benchmark, &fwdAlg);
+  cudnnConvolutionFwdAlgoPerf_t fwdAlgPerf;
+  Workspace workspace = chooseAlgorithm(dataType, args, benchmark, &fwdAlgPerf);
+
+  // update convDesc mathType since cudnn now requires both algo + mathType to figure out
+  // whether to use Tensor cores or not
+  AT_CUDNN_CHECK(cudnnSetConvolutionMathType(args.cdesc.mut_desc(), fwdAlgPerf.mathType));
```

Comment:
I'm a little confused here: why don't you modify the convolution descriptor before calling chooseAlgorithm?

Reply:
OK, this is a very good comment to put in a `Note [chooseAlgorithm doesn't respect mathType]` and reference from all of these sites :)

Reply:
Added :)